File: src/main/java/kitchenpos/menus/tobe/domain/menu/MenuNamePolicy.java
package kitchenpos.menus.tobe.domain.menu;

import kitchenpos.common.DomainService;

import java.util.Objects;

@DomainService
public class MenuNamePolicy {
    private final MenuNameProfanities menuNameProfanities;

    public MenuNamePolicy(MenuNameProfanities menuNameProfanities) {
        this.menuNameProfanities = menuNameProfanities;
    }

    public void validateName(Name displayName) {
        if (Objects.isNull(displayName.getValue()) || this.containsProfanity(displayName)) {
            throw new IllegalArgumentException();
        }
    }

    private boolean containsProfanity(Name name) {
        return menuNameProfanities.containsProfanity(name.getValue());
    }
}

File: src/main/kotlin/kitchenpos/menu/tobe/domain/MenuNamePolicy.kt
package kitchenpos.menu.tobe.domain

import kitchenpos.common.annotation.DomainService
import kitchenpos.common.domain.Profanities

@DomainService
class MenuNamePolicy(private val profanities: Profanities) {
    fun validate(name: String) {
        require(name.isNotBlank()) { "메뉴 이름은 필수값입니다." }
        require(!profanities.contains(name)) { "메뉴 이름에 금지어가 포함되어 있습니다." }
    }
}
